packages/wallet/src/Wallets/BaseWallet.ts

```diff
@@ -307,24 +371,36 @@ export class PersonalWallet implements ObservableWallet {
       filter((status) => status === ConnectionStatus.down)
     );
 
-    this.#addressDiscovery = addressDiscovery;
-    this.#addressTracker = createAddressTracker({
-      addressDiscovery$: coldObservableProvider({
-        cancel$,
-        onFatalError,
-        provider: () => addressDiscovery.discover(this.bip32Account),
-        retryBackoffConfig
-      }).pipe(
-        take(1),
-        catchError((error) => {
-          this.#logger.error('Failed to complete the address discovery process', error);
-          throw error;
-        })
-      ),
-      logger: this.#logger,
-      store: stores.addresses
-    });
-    this.addresses$ = this.#addressTracker.addresses$;
+    if (isBip32PublicCredentialsManager(this.#publicCredentialsManager)) {
+      this.#addressTracker = createAddressTracker({
+        addressDiscovery$: coldObservableProvider({
+          cancel$,
+          onFatalError,
+          provider: () => {
+            const credManager = this.#publicCredentialsManager as Bip32PublicCredentialsManager;
+            return credManager.addressDiscovery.discover(credManager.bip32Account);
+          },
+          retryBackoffConfig
+        }).pipe(
+          take(1),
+          catchError((error) => {
+            this.#logger.error('Failed to complete the address discovery process', error);
+            throw error;
+          })
+        ),
+        logger: this.#logger,
+        store: stores.addresses
+      });
+      this.addresses$ = this.#addressTracker.addresses$;
+    } else {
+      const credManager = this.#publicCredentialsManager as ScriptPublicCredentialsManager;
+      this.addresses$ = from(this.networkInfoProvider.genesisParameters()).pipe(
+        map(({ networkId }) => networkId),
+        distinctUntilChanged(),
+        switchMap((networkId) => of(getScriptAddress(credManager.paymentScript, credManager.stakingScript, networkId))),
+        map((groupedAddress) => [groupedAddress])
```

Comment:
This can be simplified now that getScriptAddress is sync
```suggestion
        map((networkId) => [getScriptAddress(credManager.paymentScript, credManager.stakingScript, networkId)])
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1111/commits/3325e052ddd9893a343e94dccbd127a5608b0ce5